### PR TITLE
Make grpc.Errorf return struct pointer

### PR DESCRIPTION
**Pull request by danruehle**
Addresses https://github.com/grpc/grpc-go/issues/698

### CODE REVIEW

**Comment by danruehle** [comment]
I signed it!

**Comment by danruehle** [comment]
@iamqizhao This is ready when you get a chance

File: call_test.go [hunk 1]
```diff
@@ -225,7 +225,7 @@ func TestInvokeLargeErr(t *testing.T) {
 	var reply string
 	req := "hello"
 	err := Invoke(context.Background(), "/foo/bar", &req, &reply, cc)
-	if _, ok := err.(rpcError); !ok {
+	if _, ok := err.(*rpcError); !ok {
```
Comment (odeke-em) [inline call_test.go]: what happens though if we get a `rpcError` or an `*rpcError`, only the `*rpcError` will be accepted. IMO we can keep compatibility by defining a small function like this
```go
func rpcErrorLike(e interface{}) bool {
  switch e.(type) {
  case *rpcError, rpcError:
     return true
  default:
     return false
  }
}
```

and then use it in those checks this way the code will be compatible with all users
so we'll change it to
```go
if !rpcErrorLike(err) {
   t.Fatalf("grpc.Invoke(_, _, _, _, _) receives non rpc error.")
}
```

Let me know what you think.

Comment (tamird) [inline call_test.go]: `rpcError` does not implement `error`, only `*rpcError` does.

Comment (odeke-em) [inline call_test.go]: ah, thanks @tamird for the correction. @danruehle never mind my comments :)

**Comment by iamqizhao** [comment]
@danruehle sorry about the delay. Can you rebase? Let's get this in soon.

**Comment by menghanl** [comment]
moved to #755

**Pull request closed without merging**